### PR TITLE
feat(googleapi): inject gax apierror.APIError into googleapi.Error

### PR DESCRIPTION
**Pull request by quartzmo**
This PR depends on googleapis/gax-go#238.

### CODE REVIEW

**Comment by quartzmo** [comment]
I ran generation locally and it succeeds:

```
➜  google-api-go-generator git:(error-wrapping) make
go build -o google-api-go-generator
./google-api-go-generator -cache=false -install -api=* -gendir=..
2022/11/02 16:59:52 Generating API abusiveexperiencereport:v1
2022/11/02 16:59:53 Generating API acceleratedmobilepageurl:v1
2022/11/02 16:59:53 Generating API accessapproval:v1
2022/11/02 16:59:54 Generating API accesscontextmanager:v1beta
2022/11/02 16:59:55 Generating API accesscontextmanager:v1
...
2022/11/02 17:05:00 Generating API workflows:v1
2022/11/02 17:05:01 Generating API youtube:v3
2022/11/02 17:05:02 Generating API youtubeAnalytics:v2
2022/11/02 17:05:02 TODO writeSchemaCode for arrays for QueryResponseRows
2022/11/02 17:05:03 Generating API youtubereporting:v1
go test . -v
=== RUN   TestMedia
--- PASS: TestMedia (0.00s)
...
    --- PASS: TestDeprecatedPkgs/no_replacement_found,_package_not_in_map (0.00s)
    --- PASS: TestDeprecatedPkgs/no_replacement_found,_version_mismatch (0.00s)
PASS
ok      google.golang.org/api/google-api-go-generator   0.987s
```

**Comment by codyoss** [comment]
@quartzmo Is this ready for review? If so, take it out of draft.

File: internal/gensupport/error.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package gensupport
+
+import (
+	"errors"
+
+	"github.com/googleapis/gax-go/v2/apierror"
+	"google.golang.org/api/googleapi"
+)
+
+// WrapError creates an apierror.APIError from err (one that
```
Comment (codyoss) [inline internal/gensupport/error.go:14]: Please use new link syntax: https://tip.golang.org/doc/comment#links

File: internal/gensupport/error.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 Google LLC. All rights reserved.
```
Comment (codyoss) [inline internal/gensupport/error.go:1]: nit: update year

File: internal/gensupport/error_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC.
```
Comment (codyoss) [inline internal/gensupport/error_test.go:1]: nit: update year

File: internal/gensupport/error.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/googleapis/gax-go/v2/apierror"
+	"google.golang.org/api/googleapi"
+)
+
+// WrapError creates an apierror.APIError from err (one that
+// does not wrap err), wraps it in err, and returns err. If
+// err is not a googleapi.Error (or a gRPC Status), it returns
+// err without modification.
+func WrapError(err error) error {
+	if apiError, ok := apierror.ParseError(err, false); ok {
```
Comment (codyoss) [inline internal/gensupport/error.go:19]: nit: reverse the logic and shift right the !ok case

File: googleapi/googleapi.go [hunk 5]
```diff
@@ -79,6 +79,8 @@ type Error struct {
 	Header http.Header
 
 	Errors []ErrorItem
+	// Err is a wrapped apierror.APIError, typically derived from this Error.
```
Comment (codyoss) [inline googleapi/googleapi.go:82]: I wonder if this should be private and we have a Wrap(error) method. Feels odd that we would say the error is a certain type but don't treat it as such. Thoughts?

**Review by codyoss — COMMENTED** [review]
Just a few small things

File: internal/gensupport/error.go (hunk 1 above)
Comment (quartzmo) [reply internal/gensupport/error.go]: Fixed, thanks!

File: internal/gensupport/error.go (hunk 2 above)
Comment (quartzmo) [reply internal/gensupport/error.go]: Fixed, thanks!

File: internal/gensupport/error_test.go (hunk 3 above)
Comment (quartzmo) [reply internal/gensupport/error_test.go]: Fixed, thanks!

File: internal/gensupport/error.go (hunk 4 above)
Comment (quartzmo) [reply internal/gensupport/error.go]: Refactored to remove a line of code and avoid early return. Is it too confusing now?

File: googleapi/googleapi.go (hunk 5 above)
Comment (quartzmo) [reply googleapi/googleapi.go:82]: This reminds me that I wanted to discuss more generally how we publicize the wrapped `APIError` to users. Ideally we would show an example of the `errors.As` usage somewhere that they can copy/paste.

Comment (quartzmo) [reply googleapi/googleapi.go]: Replaced `Error.Err` with `Error.Wrap()`

File: internal/gensupport/error.go (hunk 4 above)
Comment (codyoss) [reply internal/gensupport/error.go]: I think this is fine, but I might scoot line 20 up to top of func if I am being nitpicky. I like to see if checks directly proceeding the line the are referencing. ok is declared at 19 and checked at 21.

File: internal/gensupport/error.go [hunk 6]
```diff
@@ -11,10 +11,11 @@ import (
 	"google.golang.org/api/googleapi"
 )
 
-// WrapError creates an apierror.APIError from err (one that
-// does not wrap err), wraps it in err, and returns err. If
-// err is not a googleapi.Error (or a gRPC Status), it returns
-// err without modification.
+// WrapError creates an [apierror.APIError] from err (one that does not wrap
```
Comment (codyoss) [inline internal/gensupport/error.go:14]: nit: I think I would drop the part in () here. If makes it harder to read. I think this is implied by explicitly passing false to the func.

Comment (quartzmo) [reply internal/gensupport/error.go]: Fixed, thanks!

**Review by codyoss — APPROVED** [review]

**Comment by codyoss** [comment]
Thanks! Lets be sure to regen everything in this PR as well

**Comment by quartzmo** [comment]
Will do. I did it once and stashed the results, but I'll rebase and do a fresh run.

**Review by codyoss — APPROVED** [review]

**Pull request merged**